### PR TITLE
Checking tests

### LINKED ISSUE #11: Fix disabled tests

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #11.

### CODE REVIEW

**Review by chiaradiamarcelo — APPROVED** [review]
Looks good!

**Review by chiaradiamarcelo — CHANGES_REQUESTED** [review]
Please also include an entry in the change log file

**Comment by AnastasiiaSergienko** [comment]
I think we should also update the version?

File: src/exasol_odbc/dialect.tdd [hunk 1]
```diff
@@ -230,23 +230,23 @@
       <argument type='str' />
     </function>
     <function group='string' name='MID' return-type='str'>
-      <formula>&#10;(CASE WHEN (%2 IS NULL) OR %2 &lt; 1 THEN NULL&#10;      ELSE {fn SUBSTRING(%1,{fn FLOOR(%2)},LENGTH(%1) - ({fn FLOOR(%2)}) + 1)} END)</formula>
+      <formula>MID(%1, %2)</formula>
```
Comment (redcatbear) [inline src/exasol_odbc/dialect.tdd:233]: Null handling is missing in online handbook
https://docs.exasol.com/sql_references/functions/alphabeticallistfunctions/mid.htm
Please get that added by the tech writers. Without that I can't judge what Exasol's `MID` is supposed to do in that case and whether special care is necessary.
Also the current implementation uses `FLOOR`, indicating that floating point numbers are acceptable input. Check that `MID` behaves the same way.

File: src/exasol_odbc/dialect.tdd [hunk 2]
```diff
@@ -289,6 +289,14 @@
       <formula>UPPER(%1)</formula>
       <argument type='str' />
     </function>
+    <function group='string' name='SPACE' return-type='str'>
+      <formula>CASE WHEN %1 &lt; 1 THEN NULL WHEN %1 = 0 THEN &apos;&apos; ELSE REPEAT(&apos; &apos;,%1) END</formula>
```
Comment (redcatbear) [inline src/exasol_odbc/dialect.tdd:293]: What happens if `%1` is NULL?

File: src/exasol_odbc/dialect.tdd [hunk 3]
```diff
@@ -432,7 +444,7 @@
       <argument type='str' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>TO_CHAR(%1, &apos;Mon DD YYYY HH24:MI:SS&apos;)</formula>
+      <formula>TO_CHAR(%1)</formula>
```
Comment (redcatbear) [inline src/exasol_odbc/dialect.tdd:447]: This is more permissive than the original, so it should not break anything.

File: src/exasol_odbc/dialect.tdd [hunk 4]
```diff
@@ -952,16 +964,16 @@
       <formula part='year'>CASE WHEN %2 IS NULL OR %3 IS NULL THEN CAST(NULL AS INTEGER) ELSE (EXTRACT(YEAR FROM %3) - EXTRACT(YEAR FROM %2)) END</formula>
       <formula part='quarter'>FLOOR(MONTHS_BETWEEN(TRUNC( CAST(%3 AS DATE), &apos;Q&apos; ), TRUNC( CAST(%2 AS DATE), &apos;Q&apos; )) / 3)</formula>
       <formula part='month'>FLOOR(MONTHS_BETWEEN(TRUNC( CAST(%3 AS DATE), &apos;MONTH&apos; ), TRUNC( CAST(TO_DATE(%2, &apos;YYYY-MM-DD&apos;) AS DATE), &apos;MONTH&apos; )))</formula>
-      <formula part='week'>ABS(FLOOR((DAYS_BETWEEN(%3, &apos;1995-01-01&apos;) - MOD(7 + TO_NUMBER(TO_CHAR(%3, &apos;D&apos;))- 1, 7) - DAYS_BETWEEN(%2, &apos;1995-01-01&apos;) + MOD(7 + TO_NUMBER(TO_CHAR(%2,&apos;D&apos;))- 1, 7))/7))</formula>
+      <formula part='week'>FLOOR((DAYS_BETWEEN(%3, &apos;1995-01-01&apos;) - MOD(7 + TO_NUMBER(TO_CHAR(%3, &apos;D&apos;))- 1, 7) - DAYS_BETWEEN(%2, &apos;1995-01-01&apos;) + MOD(7 + TO_NUMBER(TO_CHAR(%2,&apos;D&apos;))- 1, 7))/7)</formula>
```
Comment (redcatbear) [inline src/exasol_odbc/dialect.tdd:967]: Bold move removing the `ABS`. This breaks existing behavior. Please check with the Tableau team, if they expect only positive numbers here.

File: src/exasol_odbc/dialect.tdd [hunk 5]
```diff
@@ -952,16 +964,16 @@
...
       <formula part='minute'>MINUTES_BETWEEN(%3, %2)</formula>
-      <formula part='second'>SECONDS_BETWEEN(%3, %2)</formula>
+      <formula part='second'>CAST(SECONDS_BETWEEN(%3, %2) AS INTEGER)</formula>
       <argument type='localstr' />
       <argument type='datetime' />
       <argument type='datetime' />
     </date-function>
     <date-function name='DATEDIFF' return-type='int'>
-      <formula part='week'>ABS(FLOOR((DAYS_BETWEEN(%3, &apos;1995-01-01&apos;) - MOD(7 + TO_NUMBER(TO_CHAR(%3,&apos;D&apos;))- %4, 7) - DAYS_BETWEEN(%2, &apos;1995-01-01&apos;) + MOD(7 + TO_NUMBER(TO_CHAR(%2,&apos;D&apos;))- %4, 7))/7))</formula>
+      <formula part='week'>FLOOR((DAYS_BETWEEN(%3, &apos;1995-01-01&apos;) - MOD(7 + TO_NUMBER(TO_CHAR(%3,&apos;D&apos;))- %4, 7) - DAYS_BETWEEN(%2, &apos;1995-01-01&apos;) + MOD(7 + TO_NUMBER(TO_CHAR(%2,&apos;D&apos;))- %4, 7))/7)</formula>
```
Comment (redcatbear) [inline src/exasol_odbc/dialect.tdd:976]: Same here.

File: src/exasol_odbc/dialect.tdd [hunk 6]
```diff
@@ -952,16 +964,16 @@
...
       <formula part='quarter'>FLOOR(MONTHS_BETWEEN(TRUNC( CAST(%3 AS DATE), &apos;Q&apos; ), TRUNC( CAST(%2 AS DATE), &apos;Q&apos; )) / 3)</formula>
       <formula part='month'>FLOOR(MONTHS_BETWEEN(TRUNC( CAST(%3 AS DATE), &apos;MONTH&apos; ), TRUNC( CAST(TO_DATE(%2, &apos;YYYY-MM-DD&apos;) AS DATE), &apos;MONTH&apos; )))</formula>
-      <formula part='week'>ABS(FLOOR((DAYS_BETWEEN(%3, &apos;1995-01-01&apos;) - MOD(7 + TO_NUMBER(TO_CHAR(%3, &apos;D&apos;))- 1, 7) - DAYS_BETWEEN(%2, &apos;1995-01-01&apos;) + MOD(7 + TO_NUMBER(TO_CHAR(%2,&apos;D&apos;))- 1, 7))/7))</formula>
+      <formula part='week'>FLOOR((DAYS_BETWEEN(%3, &apos;1995-01-01&apos;) - MOD(7 + TO_NUMBER(TO_CHAR(%3, &apos;D&apos;))- 1, 7) - DAYS_BETWEEN(%2, &apos;1995-01-01&apos;) + MOD(7 + TO_NUMBER(TO_CHAR(%2,&apos;D&apos;))- 1, 7))/7)</formula>
       <formula part='hour'>HOURS_BETWEEN(%3, %2)</formula>
       <formula part='minute'>MINUTES_BETWEEN(%3, %2)</formula>
-      <formula part='second'>SECONDS_BETWEEN(%3, %2)</formula>
+      <formula part='second'>CAST(SECONDS_BETWEEN(%3, %2) AS INTEGER)</formula>
```
Comment (redcatbear) [inline src/exasol_odbc/dialect.tdd:970]: Wouldn't `FLOOR` be clearer here?

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: src/exasol_odbc/dialect.tdd (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/exasol_odbc/dialect.tdd:967]: But this is what their tests expect. After I removed ABS, the tests passed

File: src/exasol_odbc/dialect.tdd (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/exasol_odbc/dialect.tdd:293]: It returns null

File: src/exasol_odbc/dialect.tdd (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/exasol_odbc/dialect.tdd:233]: When one of the values is null, then the function returns null

File: src/exasol_odbc/dialect.tdd (hunk 4 above)
Comment (chiaradiamarcelo) [reply src/exasol_odbc/dialect.tdd:967]: According to the tests expectations, the don't expect only positive numbers, but both positive and negative. We can double  check with the Tableau team anyway.

File: tdvt/config/exasol_odbc.ini [hunk 7]
```diff
@@ -4,21 +4,11 @@ LogicalQueryFormat = TESTV1
...
-#string.space: incorrect results
-#date.datediff.week: absolute value instead of expected
-#date.datediff.sow.week: absolute value instead of expected
-#string.mid: incorrect results
-#string.isdate: not supported?
 #calcs_data: not supported
-ExpressionExclusions_Standard = cast.str.date,date.cast,date.datediff.second,date.datediff.sow.second,date.datepart.second,date.datepart.sow.second,math.div,string.space,string.split,date.datediff.week,date.datediff.sow.week,string.mid,string.isdate, calcs_data
+#string.split: this function is not supported in Exasol
+#string.space.int.constant and string.space.int.var:  incorrect results, can't find the problem with thespace. Generated query works fine in the SQL client 
+#string.mid.calc: incorrect results,can't find the problem, MID in Tableau and Exasol seem to be identical
```
Comment (chiaradiamarcelo) [inline tdvt/config/exasol_odbc.ini:10]:
```suggestion
```

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**